habitat_sim/simulator.py

```diff
@@ -25,12 +25,17 @@
 from habitat_sim.bindings import cuda_enabled
 from habitat_sim.logging import logger
 from habitat_sim.nav import GreedyGeodesicFollower, NavMeshSettings, PathFinder
-from habitat_sim.sensor import SensorType
+from habitat_sim.sensor import SensorSpec, SensorType
 from habitat_sim.sensors.noise_models import make_sensor_noise_model
 from habitat_sim.sim import SimulatorBackend, SimulatorConfiguration
 from habitat_sim.utils.common import quat_from_angle_axis
 
 
+def _overwrite_sensor_spec_uuid(spec: SensorSpec, agent_id: int = 0) -> SensorSpec:
```

Comment:
@erikwijmans Is overwriting the UUID like this for multi-agents the way to go?

Reply:
Maybe I should make this method a class function so that it can be overrided?

Reply:
IMO returning a list of observation-dicts makes more sense for multi-agent

Reply:
@erikwijmans Yeah, I just wanted to minimize API breakage between Habitat-API and Habitat-Lab for now. This is for the use case for having multiple static sensors around the scene (IE. Security Cameras) or alternatively agents that are moved outside of step.

Reply:
But the question was specifically "is overwriting the UUID like this for multi-agents the way to go?".

We can always still return just a dict when there is a single agent.

Reply:
While UUID implies UNIQUE to me, meaning that any sensor regardless of agent should have a UNIQUE UID.

Reply:
Since the UUID is part of the config though, it prevents a config from being reused for multiple agents so some UUID mutation has / should occur in my opinion.

Reply:
Maybe the UUID overwriting could be done in Habitat-Lab?

Reply:
I disagree that it has to be globally unique.  It has to be unique for that agent, but I don't see what it needs to be globally unique.

Reply:
@erikwijmans I just started the redesign, but our call guarantees get quite messy but doing this. Really don't like returning different types like that...

Reply:
Okay, I think I cleaned it up quite a bit.